var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -218,12 +218,8 @@ def cmake_args(self):
             ])
         else:
             prefix = spec['opengl'].prefix
-
             opengl_include_dir = prefix.include
-            opengl_library = os.path.join(prefix.lib, 'libGL.so')
-            if 'darwin' in spec.architecture:
-                opengl_include_dir = prefix
-                opengl_library = prefix
+            opengl_library = spec['opengl'].libs.joined()
```

Comment:
This needs to be using the `gl` virtual package instead of `opengl`.  `opengl` is a dummy package intended to let the system-provided OpenGL drivers to be be used.  `gl` is a virtual package provided by both `mesa` and `opengl` and is what downstream packages should be depending on.